### PR TITLE
Vector support

### PR DESCRIPTION
**Pull request by BrettRToomey**
Just a local copy of #103 with updates

### CODE REVIEW

File: Sources/LLVM/IRType.swift [hunk 1]
```diff
@@ -36,6 +36,16 @@ public extension IRType {
   }
 }
 
+extension IRType {
+    internal func lowerFromVector() -> IRType {
```
Comment (CodaFi) [inline Sources/LLVM/IRType.swift]: I don't anticipate this being used outside of the IRBuilder, so a private free function would work.

Comment (BrettRToomey) [reply Sources/LLVM/IRType.swift]: Sure, I head home in a bit and then I can update the PR.

**Comment by CodaFi** [comment]
Last thing: Could you add file check tests for the vector instructions?

**Comment by CodaFi** [comment]
Eh, I've got tests.  

⛵️

**Pull request merged**